components/my-account/Lists/Lists.tsx

```diff
@@ -0,0 +1,148 @@
+    pageSize: publicRuntimeConfig.b2bList.pageSize,
+    sortBy: publicRuntimeConfig.b2bList.sortBy,
+    filter: publicRuntimeConfig.b2bList.filter,
+  })
+
+  // screen size declared
+  const theme = useTheme()
+  const mdScreen = useMediaQuery<boolean>(theme.breakpoints.up('md'))
+
+  const { t } = useTranslation('common')
+
+  // on changing page number
+  useEffect(() => {
+    const newStartIndex = state.pageSize * (page - 1)
+    setState({ ...state, startIndex: newStartIndex })
+  }, [page])
+
+  // fetching wishlist data
+  const { data, isPending } = useGetWishlist(state)
+
+  // this will be used for creating new list
+  const { user } = useAuthContext()
+  const userId = user?.userId
+
+  function filterChange() {
+    console.log(
+      'WIP: Change filter for list create by me only if true ==> "createBy eq " + userId, else ==> ""'
+    )
+  }
+  if (!isPending) {
+    const rows = data.items.map((item: CrWishlist) => {
+      return {
+        ...item,
+        createDate: item.auditInfo && getDate(item.auditInfo.createDate),
+        createBy: item.auditInfo && item.auditInfo.createBy,
+      }
+    })
+
+    const totalCount = data.totalCount
+    const pageCount = parseInt(
+      (totalCount % state.pageSize === 0
+        ? totalCount / state.pageSize
+        : totalCount / state.pageSize + 1
+      ).toString()
+    )
+    return (
+      <Box style={{ padding: '10px 10px 10px 0' }}>
+        <FormControlLabel
+          label={t('show-only-my-lists')}
+          control={<Checkbox onChange={filterChange} sx={{ fontSize: '16px' }} />}
+        />
+        {!mdScreen && (
+          <>
+            <FormControl
+              sx={{
+                width: '100%',
+                borderBottom: 'none',
+              }}
+            >
+              <Input
+                placeholder={t('search-by-name')}
+                sx={{
+                  height: '32px',
+                  fontSize: '14px',
+                  padding: '8px 12px',
+                  border: '1px solid #cdcdcd',
+                  borderRadius: '4px',
+                  width: '100%',
+                  margin: '16px 0',
+                  borderBottom: 'none',
+                }}
+                startAdornment={
+                  <InputAdornment position="start">
+                    <Search />
+                  </InputAdornment>
+                }
+              />
+            </FormControl>
+          </>
+        )}
+        <ListTable rows={rows} />
+        <Pagination
+          count={pageCount}
+          shape={`rounded`}
+          size="small"
+          sx={{ marginTop: '15px' }}
+          onChange={(event: ChangeEvent<any>, page: number) => {
+            setPage(page)
+          }}
+        />
+        <ReactQueryDevtools initialIsOpen={false} />
+      </Box>
+    )
+  } else {
+    return (
+      <div style={{ textAlign: 'center' }}>
+        <CircularProgress />
+      </div>
+    )
+  }
+}
+
+export default Lists
```

Comment:
**1. Story missing**

**2. Testcases are missing**

**3. Remove devtool code**
   ```
<ReactQueryDevtools initialIsOpen={false} />
   import { ReactQueryDevtools } from '@tanstack/react-query-devtools'
```
   
**4. Instead of giving generic name give specific names to state**
   c`onst [state, setState] = useState<ListsState>`
   should be  `const [paginationState, setPaginationState] = useState`
   
**5. Avoid using useEffect for syncing state**
```
  useEffect(() => {
    const newStartIndex = state.pageSize * (page - 1)
    setState({ ...state, startIndex: newStartIndex })
  }, [page]) 
```
  remove code from useEffect
  
**6. Logical group all the code**
   const theme = useTheme()
   const mdScreen = useMediaQuery<boolean>(theme.breakpoints.up('md'))
   const { t } = useTranslation('common')
   const { user } = useAuthContext()
  
**7. prefer using arrow functions**
   function filterChange()
   should be const handleFilterChange = () => {}

**8. Instead of using big if(){} else{} use only if(){}**   
  ```
 if (isPending) return <CircularProgress />
   return ()
```
  
**9. Instead of calculating  const pageCount = parseInt(), fetch it from backend**
   
**10. No need of  const totalCount = data.totalCount, you can directly use  data.totalCount**

**11. Move below function to lib/helpers**
    function getDate(date: number) {}

**12. Define below interface at common place and reuse it (between users and list)**
    interface ListsState {
	  startIndex: number
	  pageSize: number
	  filter: string
	  sortBy: string
	}

**13. Define SX at top and use it**
    {
	  height: '32px',
	  fontSize: '14px',
	  padding: '8px 12px',
	  border: '1px solid #cdcdcd',
	  borderRadius: '4px',
	  width: '100%',
	  margin: '16px 0',
	  borderBottom: 'none',
	}
	
**14. Separate JSX code and logic**
 onChange={(event: ChangeEvent<any>, page: number) => {
            setPage(page)
          }}
		  
  should be onChange = {handlePagination}